src/orion/algo/tpe.py

```diff
@@ -0,0 +1,351 @@
+
+        super(TPE, self).__init__(space,
+                                  seed=seed,
+                                  n_initial_points=n_initial_points,
+                                  n_ei_candidates=n_ei_candidates,
+                                  gamma=gamma,
+                                  equal_weight=equal_weight,
+                                  prior_weight=prior_weight)
+
+        for dimension in self.space.values():
+
+            if dimension.type not in ['real']:
+                raise ValueError("TPE now only supports Real Dimension.")
+
+            if dimension.prior_name not in ['uniform']:
+                raise ValueError("TPE now only supports uniform as prior.")
+
+            shape = dimension.shape
+            if shape and len(shape) != 1:
+                raise ValueError("TPE now only supports 1D shape.")
+
+        self.seed_rng(seed)
+
+    def seed_rng(self, seed):
+        """Seed the state of the random number generator.
+
+        :param seed: Integer seed for the random number generator.
+        """
+        self.rng = numpy.random.RandomState(seed)
+
+    @property
+    def state_dict(self):
+        """Return a state dict that can be used to reset the state of the algorithm."""
+        _state_dict = super(TPE, self).state_dict
+
+        _state_dict['rng_state'] = self.rng.get_state()
+        _state_dict['seed'] = self.seed
+        return _state_dict
+
+    def set_state(self, state_dict):
+        """Reset the state of the algorithm based on the given state_dict
+
+        :param state_dict: Dictionary representing state of an algorithm
+        """
+        super(TPE, self).set_state(state_dict)
+
+        self.seed_rng(state_dict['seed'])
+        self.rng.set_state(state_dict['rng_state'])
+
+    def suggest(self, num=1):
+        """Suggest a `num` of new sets of parameters. Randomly draw samples
+        from the import space and return them.
+
+        :param num: how many sets to be suggested.
+
+        .. note:: New parameters must be compliant with the problem's domain
+           `orion.algo.space.Space`.
+        """
+        if num > 1:
+            raise ValueError("TPE should suggest only one point.")
+
+        samples = []
+        if len(self._trials_info) < self.n_initial_points:
+            new_point = self.space.sample(1, seed=tuple(self.rng.randint(0, 1000000, size=3)))[0]
+            samples.append(new_point)
+        else:
+            point = []
+            below_points, above_points = self.split_trials()
+            below_points = numpy.array([flatten_dims(point, self.space) for point in below_points])
+            above_points = numpy.array([flatten_dims(point, self.space) for point in above_points])
+            idx = 0
+            for dimension in self.space.values():
+
+                if dimension.type == 'real':
+                    shape = dimension.shape
+                    if not shape:
```

Comment:
Maybe we could keep this for when we support shape > 1.

Reply:
did not quite get this one, we support shape value > 1, but only 1D shape.

Reply:
Ah! My bad, I misread that tpe only supported shape == 1 sor far but latter I was confused because your TPE implementation supports 1d arrays. Great! :)